source/MaterialXView/ShaderValidators/Osl/OslValidator.h

```diff
@@ -106,13 +154,31 @@ class OslValidator : public ShaderValidator
     /// @param oslFileName Name of OSL file
     void compileOSL(const std::string& oslFileName);
 
+    ///
+    /// Shade using OSO input file. Return errors in result string 
+    /// @param shaderName Name of OSL shader. A corresponding .oso file is assumed to exist
+    /// @param outputName Name of OSL shader output to use.
+    void shadeOSL(const std::string& shaderName, const std::string& outputName);
+
     /// Constructor
     OslValidator();
 
   private:
+    /// "oslc" executable name
     std::string _oslCompilerExecutable;
+    /// OSL include path name
     std::string _oslIncludePathString;
+    /// Output file path. File name does not include an extension
     std::string _oslOutputFilePathString;
+
+    /// "testshade" executable name
```

Comment:
More input required rendering. We need the exe + we need the actual shader output name,
Niklas will look at providing outputs in the Shader output from ShaderGen but this is not available yet so need
to explicitly be set via interfaces here.